gadget/device/encrypt.go

```diff
@@ -164,15 +166,16 @@ const (
 
 // VolumesAuthOptions contains options for the volumes authentication
 // mechanism (e.g. passphrase authentication).
-//
-// TODO: Add PIN option when secboot support lands.
 type VolumesAuthOptions struct {
 	Mode       AuthMode      `json:"mode,omitempty"`
 	Passphrase string        `json:"passphrase,omitempty"`
+	PIN        string        `json:"pin,omitempty"`
 	KDFType    string        `json:"kdf-type,omitempty"`
 	KDFTime    time.Duration `json:"kdf-time,omitempty"`
 }
 
+var validPIN = regexp.MustCompile(`^[0-9]+$`)
+
 // Validates authentication options.
 func (o *VolumesAuthOptions) Validate() error {
```

Comment:
does this check that PIN and Passhprase are not set at the same time?

Reply:
`VolumesAuthOptions.Mode` is used to determine which field to look at, I cannot think of scenarios where checking they are not set at the same time would be useful. I can add the check for good measure anyway.